public/app/plugins/datasource/prometheus/querybuilder/components/PromQueryBuilder.tsx

```diff
@@ -108,7 +110,15 @@ export const PromQueryBuilder = React.memo<Props>(({ datasource, query, onChange
           onChange={onChange}
           onRunQuery={onRunQuery}
         />
-        <PromQueryBuilderHints datasource={datasource} query={query} onChange={onChange} data={data} />
+        <QueryBuilderHints
+          datasource={datasource}
+          query={query}
+          // eslint-disable-next-line
+          onChange={onChange as (update: PromLokiVisualQuery) => void}
```

Comment:
I don't like this, but not sure how else to solve it. Maybe you have better idea. The issue is that `metric` is in `PromLokiVisualQuery` optional but in `PromVisualQuery` it is required. 
```
Types of property 'metric' are incompatible.
        Type 'string | undefined' is not assignable to type 'string'.
          Type 'undefined' is not assignable to type 'string'.
```

Reply:
Resolved by correctly typing